pkg/api/defaults.go

```diff
@@ -692,29 +692,12 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 	}
 }
 
-// setWindowsProfileDefaults sets default WindowsProfile values
+// setWindowsProfileDefaults sets default Windows distro
 func (p *Properties) setWindowsProfileDefaults(isUpgrade, isScale bool) {
 	windowsProfile := p.WindowsProfile
 	if !isUpgrade && !isScale {
-		if windowsProfile.WindowsPublisher == "" {
-			windowsProfile.WindowsPublisher = AKSWindowsServer2019OSImageConfig.ImagePublisher
-		}
-		if windowsProfile.WindowsOffer == "" {
-			windowsProfile.WindowsOffer = AKSWindowsServer2019OSImageConfig.ImageOffer
-		}
-		if windowsProfile.WindowsSku == "" {
-			windowsProfile.WindowsSku = AKSWindowsServer2019OSImageConfig.ImageSku
```

Comment:
Please keep this logic in here if windowsProfile.Distro is not set.
Our documented guidance to customers is to set WindowsPublisher, WindowsOffer, and WindowsSku if they want to get the lastest tested version of an image.
https://github.com/Azure/aks-engine/blob/master/docs/topics/clusterdefinitions.md#marketplace-images

If we remove this logic here ImageVersion users following this guidance will likely be broken.

Reply:
@marosset The document may not be accurate. Even if users set `windowsPublisher`, `windowsOffer`, and `windowsSku`, the `latest` version will not be used when the values of those 3 variables equal to the values of `AKSWindowsServer2019OSImageConfig` or `WindowsServer2019OSImageConfig`.
Different Azure environments may have different default OS images so we may need to set default values by Azure environments.
I prefer to set `ImageVersion` to `latest` if all of `windowsPublisher`, `windowsOffer`, and `windowsSku` are specified but `ImageVersion` is empty in setWindowsProfileDefaults. And we can update the doc in the link you provided. How do you think?

Reply:
The old code supports to set `latest` so I just leave the code there in this PR. If we do not want users to use `latest`, we can remove it later.